developer/src/kmc-convert/src/convert-utils.ts

```diff
@@ -0,0 +1,83 @@
+
+/*
+ * Keyman is 2025 copyright (C) SIL International. MIT License.
+ *
+ * Created by S. Schmitt on 2025-08-24
+ *
+ * functions for kmc-convert utils
+ *
+ */
+
+/**
+ * @brief  function to convert a numeric character reference or a unicode value to a unicode character e.g. &#x63 -> c;  U+1F60E -> 😎
+ * @param  inputString the value that will converted
+ * @return a unicode character like 'c', 'ሴ', '😎' or undefined if inputString is not recognized
+ */
+export function convertToUnicodeCharacter(inputString: string): string {
+
+  if ((inputString === null) || (inputString === undefined)) {
+    return undefined;
+  }
+
+  // e.g. U+0061 U+1234 U+1F60E
+  else if (inputString.match(/^U\+([0-9a-f]{2,6})$/i)) {
+    return String.fromCodePoint(parseInt((inputString.match(/^U\+([0-9a-f]{2,6})$/i))[1], 16));
+  }
+
+  // e.g. &#x61;  &#x1234; &#x1F60E;
+  else if (inputString.match(/^&#x([0-9a-f]{2,6});$/i)) {
+    return String.fromCodePoint(parseInt((inputString.match(/^&#x([0-9a-f]{2,6});$/i))[1], 16));
+  }
+
+  // e.g. &#97; &#4660; &#128518;
+  else if (inputString.match(/^&#([0-9a-f]{2,6});$/i)) {
+    return String.fromCodePoint(parseInt((inputString.match(/^&#([0-9a-f]{2,6});$/i))[1], 10));
+  }
+
+  // e.g. &gt; &quot;
+  else if (inputString.match(/^&([a-z]{1,4});$/i)) {
+    if (inputString === '&gt;') { return '>'; }
+    else if (inputString === '&lt;') { return '<'; }
+    else if (inputString === '&amp;') { return '&'; }
+    else if (inputString === '&apos;') { return "'"; }
+    else if (inputString === '&quot;') { return '"'; }
+    else return undefined;
+  }
```

Comment:
This is a little tricky. A named entity can have an arbitary length. Best not to restrict it to 1-4 characters.